runtime/src/main/java/org/corfudb/util/Utils.java

```diff
@@ -228,10 +204,32 @@ public static long getLogTail(Layout layout, CorfuRuntime runtime) {
         } else if (segment.getReplicationMode() == Layout.ReplicationMode.QUORUM_REPLICATION) {
             throw new UnsupportedOperationException();
         }
-
         return globalLogTail;
     }
 
+    public static long getLogSize(Layout layout, CorfuRuntime runtime, long startAddress, long endAddress) {
+        long size = 0;
+        long segmentSize = 0;
+        long currentEnd = 0;
+        for (Layout.LayoutSegment segment : layout.getSegments()) {
+            if (segment.getEnd() > 0 && segment.getEnd() <= startAddress || endAddress <= segment.getStart())
+                    continue;
+
+            currentEnd = endAddress;
+            if (segment.getEnd() > 0)
+                currentEnd = Long.min(endAddress, segment.getEnd() -1);
+
+            Layout.LayoutStripe stripe = segment.getStripes().get(0);
+            size += CFUtils.getUninterruptibly (runtime.getLayoutView()
+                    .getRuntimeLayout (layout)
+                    .getLogUnitClient (stripe.getLogServers().get(DEFAULT_LOGUNIT))
```

Comment:
I think its better to avoid this term, because there is no "default logunit". It is protocol dependent. For chain replication, it should be the head node and that should be explicit in the logic. Also, the replication protocol needs to be Chain Replication for this logic to work.